From beacf69d2e4ed2d302f17e1ee17ca40b597d2d3f Mon Sep 17 00:00:00 2001
From: Andy Yan <andy.yan@rock-chips.com>
Date: Fri, 10 Aug 2018 09:11:30 +0800
Subject: [PATCH] arm: rk3308_linux_aarch32[_debug]_defconfig: enable zram
 support

Zram allows Linux to make more efficient use of RAM, since
the system can then hold more pages of memory in the compressed
swap than if the same amount of RAM had been used as application
memory or disk cache. This is particularly effective on machines
that do not have much ram space.

Enable:
echo 16M > /sys/block/zram0/disksize
mkswap /dev/zram0
swapon /dev/zram0

More detial: Documentation/blockdev/zram.txt

Change-Id: I23954dbc7df79d4faac02cf7f393dbfc37858ee6
Signed-off-by: Andy Yan <andy.yan@rock-chips.com>
---
 arch/arm/configs/rk3308_linux_aarch32_debug_defconfig | 5 +++--
 arch/arm/configs/rk3308_linux_aarch32_defconfig       | 5 +++--
 2 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/arch/arm/configs/rk3308_linux_aarch32_debug_defconfig b/arch/arm/configs/rk3308_linux_aarch32_debug_defconfig
index 2145ff8f4ebc..3e57addaebbe 100644
--- a/arch/arm/configs/rk3308_linux_aarch32_debug_defconfig
+++ b/arch/arm/configs/rk3308_linux_aarch32_debug_defconfig
@@ -1,7 +1,6 @@
 # CONFIG_LOCALVERSION_AUTO is not set
 CONFIG_KERNEL_LZ4=y
 CONFIG_DEFAULT_HOSTNAME="localhost"
-# CONFIG_SWAP is not set
 CONFIG_SYSVIPC=y
 CONFIG_FHANDLE=y
 CONFIG_NO_HZ=y
@@ -39,6 +38,7 @@ CONFIG_THUMB2_KERNEL=y
 # CONFIG_CPU_SW_DOMAIN_PAN is not set
 # CONFIG_COMPACTION is not set
 CONFIG_DEFAULT_MMAP_MIN_ADDR=32768
+CONFIG_ZSMALLOC=y
 CONFIG_UACCESS_WITH_MEMCPY=y
 # CONFIG_SWIOTLB is not set
 CONFIG_CMDLINE="coherent_pool=4k user_debug=31"
@@ -83,7 +83,8 @@ CONFIG_RFKILL=y
 CONFIG_DEVTMPFS=y
 CONFIG_DEVTMPFS_MOUNT=y
 # CONFIG_ALLOW_DEV_COREDUMP is not set
-# CONFIG_BLK_DEV is not set
+CONFIG_ZRAM=y
+CONFIG_ZRAM_LZ4_COMPRESS=y
 CONFIG_NETDEVICES=y
 # CONFIG_NET_CORE is not set
 # CONFIG_ETHERNET is not set
diff --git a/arch/arm/configs/rk3308_linux_aarch32_defconfig b/arch/arm/configs/rk3308_linux_aarch32_defconfig
index b5b02a4ccfd5..fbacc3d30f30 100644
--- a/arch/arm/configs/rk3308_linux_aarch32_defconfig
+++ b/arch/arm/configs/rk3308_linux_aarch32_defconfig
@@ -1,7 +1,6 @@
 # CONFIG_LOCALVERSION_AUTO is not set
 CONFIG_KERNEL_LZ4=y
 CONFIG_DEFAULT_HOSTNAME="localhost"
-# CONFIG_SWAP is not set
 CONFIG_SYSVIPC=y
 CONFIG_FHANDLE=y
 CONFIG_NO_HZ=y
@@ -39,6 +38,7 @@ CONFIG_THUMB2_KERNEL=y
 # CONFIG_CPU_SW_DOMAIN_PAN is not set
 # CONFIG_COMPACTION is not set
 CONFIG_DEFAULT_MMAP_MIN_ADDR=32768
+CONFIG_ZSMALLOC=y
 CONFIG_UACCESS_WITH_MEMCPY=y
 # CONFIG_SWIOTLB is not set
 CONFIG_CMDLINE="coherent_pool=4k loglevel=0"
@@ -83,7 +83,8 @@ CONFIG_RFKILL=y
 CONFIG_DEVTMPFS=y
 CONFIG_DEVTMPFS_MOUNT=y
 # CONFIG_ALLOW_DEV_COREDUMP is not set
-# CONFIG_BLK_DEV is not set
+CONFIG_ZRAM=y
+CONFIG_ZRAM_LZ4_COMPRESS=y
 CONFIG_NETDEVICES=y
 # CONFIG_NET_CORE is not set
 # CONFIG_ETHERNET is not set
-- 
2.35.3

